js/latoken.js

```diff
@@ -0,0 +1,580 @@
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'low': this.safeFloat (ticker, 'low'),
+            'high': this.safeFloat (ticker, 'high'),
+            'bid': undefined,
+            'bidVolume': undefined,
+            'ask': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': open,
+            'close': this.safeFloat (ticker, 'close'),
+            'last': undefined,
+            'previousClose': undefined,
+            'change': priceChange,
+            'percentage': priceChangePercent,
+            'average': undefined,
+            'baseVolume': this.safeFloat (ticker, 'volume'),
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let response = await this.publicGetMarketDataTicker (this.extend ({
+            'symbol': market['symbol'],
+        }, params));
+        return this.parseTicker (response, market);
+    }
+
+    async fetchCurrencies (params = {}) {
+        if (Object.keys (params).length) {
+            const currencies = await this.publicGetExchangeInfoCurrencies (params);
+            let id = currencies['currencyId'];
+            let symbol = currencies['symbol'];
+            let name = currencies['name'];
+            let precision = currencies['precission'];
+            let type = currencies['type'];
+            let fee = currencies['fee'];
+            return {
+                'id': id,
+                'symbol': symbol,
+                'name': name,
+                'precision': precision,
+                'type': type,
+                'fee': fee,
+            };
+        }
+        const currencies = await this.publicGetExchangeInfoCurrencies ();
+        let result = [];
+        for (let i = 0; i < currencies.length; i++) {
+            let currency = currencies[i];
+            let id = currency['currencyId'];
+            let symbol = currency['symbol'];
+            let name = currency['name'];
+            let precision = currency['precission'];
+            let type = currency['type'];
+            let fee = currency['fee'];
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'name': name,
+                'precision': precision,
+                'type': type,
+                'fee': fee,
+            });
+        }
+        return result;
+    }
+
+    parseTrade (trade, market = undefined) {
+        let info = trade;
+        let type = undefined;
+        let fee = undefined;
+        return {
+            'info': info,
+            'timestamp': trade.timestamp,
+            'datetime': this.iso8601 (trade.timestamp),
+            'symbol': trade.symbol,
+            'pairId': trade.pairId,
+            'id': trade.id,
+            'order': trade.orderId,
```

Comment:
We're still violating dictionary keys access `.xxx` vs `['xxx']` here, need to make sure that we don't to this – throughout this entire file.

Reply:
done